apps/prairielearn/src/ee/institution/admin/courseInstance.ts

```diff
@@ -0,0 +1,39 @@
+import { Router } from 'express';
+import { z } from 'zod';
+import asyncHandler = require('express-async-handler');
+import { loadSqlEquiv, queryRow } from '@prairielearn/postgres';
+import { getInstitution } from '../utils';
+import { CourseInstanceSchema, CourseSchema } from '../../../lib/db-types';
+import { InstitutionAdminCourseInstance } from './courseInstance.html';
+
+const sql = loadSqlEquiv(__filename);
+const router = Router({ mergeParams: true });
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    const institution = await getInstitution(req.params.institution_id);
+    const { course, course_instance } = await queryRow(
+      sql.select_course_and_instance,
+      {
+        institution_id: req.params.institution_id,
+        course_id: req.params.course_id,
+        course_instance_id: req.params.course_instance_id,
+      },
+      z.object({
+        course: CourseSchema,
+        course_instance: CourseInstanceSchema,
+      })
+    );
```

Comment:
In this case we could avoid all SQL for the page by using `selectAndAuthz...` functions. Not pushing you to switch to this model, but rather just pointing out how they would work.